ci/jenkinsfiles/common-lib.groovy

```diff
@@ -349,10 +349,20 @@ void uploadPackage(String version, String credential,  String connectUrl) {
   withCredentials(
     [usernameColonPassword(credentialsId: "${credential}", variable: 'CONNECT_PASS')]
   ) {
-    sh """
-      PACKAGE="nuxeo-retention-package/target/nuxeo-retention-package-${version}.zip"
-      curl -i -u "$CONNECT_PASS" -F package=@\$PACKAGE "$connectUrl"/site/marketplace/upload?batch=true
-    """
+    def status = sh(
+      returnStatus: true,
+      script:
+        """
+          PACKAGE="nuxeo-retention-package/target/nuxeo-retention-package-${version}.zip"
+          curl -i -u "$CONNECT_PASS" -F package=@\$PACKAGE "$connectUrl"/site/marketplace/upload?batch=true | grep 'HTTP/1.1 200'
```

Comment:
Can we try to use the curl params https://curl.se/docs/manpage.html 
for example: 
`curl -o /dev/null -s -w "%{http_code}\n" https://www.google.com`

`-o` is to not get the whole output and send it to `/dev/null`
`-s` is silent 

`http_code` _The numerical response code that was found in the last retrieved HTTP(S) or FTP(s) transfer. In 7.18.2 the alias response_code was added to show the same info._

Also note that we can check other Successful Codes (2xx)


200 OK. 
201 Created. 
202 Accepted. 
204 No Content.

We can check or ask the NOS team to be sure about the http status, if it can be different than `200`

Reply:
@RSalem07 ,
Nice idea 💡 , I will apply the changes ( and also check with NOS Team, if they have implemented something specific concerning the returned code value).